devices/Atgm336h/Atgm336h.cs

```diff
@@ -0,0 +1,233 @@
+    {
+        /// <summary>
+        /// Delegate type to handle the event when the GPS module fix status changes.
+        /// </summary>
+        /// <param name="fix">The new fix status.</param>
+        public delegate void FixChangedHandler(Fix fix);
+
+        /// <summary>
+        /// Delegate type to handle the event when the GPS module mode changes.
+        /// </summary>
+        /// <param name="mode">The new GPS module mode.</param>
+        public delegate void ModeChangedHandler(Mode mode);
+
+        /// <summary>
+        /// Delegate type to handle the event when the GPS module location changes.
+        /// </summary>
+        /// <param name="latitude">The new latitude.</param>
+        /// <param name="longitude">The new longitude.</param>
+        public delegate void LocationChangeHandler(double latitude, double longitude);
+        
+        /// <summary>
+        /// Gets the fix status of the GPS module.
+        /// </summary>
+        public Fix Fix { get; private set; } = Fix.NoFix;
+
+        /// <summary>
+        /// Gets the mode of the GPS module.
+        /// </summary>
+        /// <value>
+        /// The mode of the GPS module.
+        /// </value>
+        public Mode Mode { get; private set; } = Mode.Unknown;
+
+        /// <summary>
+        /// Gets the latitude value.
+        /// </summary>
+        public double Latitude { get; private set; }
+
+        /// <summary>
+        /// Gets the longitude value.
+        /// </summary>
+        public double Longitude { get; private set; }
+
+        /// <summary>
+        /// Represents the event handler for when the fix status of the GPS module changes.
+        /// </summary>
+        public event FixChangedHandler FixChanged;
+
+        /// <summary>
+        /// Event that occurs when the location changes.
+        /// </summary>
+        public event LocationChangeHandler LocationChanged;
+
+        /// <summary>
+        /// Represents the event that is raised when the mode of the GPS module is changed.
+        /// </summary>
+        public event ModeChangedHandler ModeChanged;
+        
+        private readonly bool _shouldDispose;
+        
+        private SerialPort _serialDevice;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Atgm336h" /> class.
+        /// </summary>
+        /// <param name="serialPort">Serial port name.</param>
+        public Atgm336h(string serialPort)
+        {
+            _serialDevice = new SerialPort(serialPort);
+            _serialDevice.BaudRate = 9600;
+            _serialDevice.Parity = Parity.None;
+            _serialDevice.StopBits = StopBits.One;
+            _serialDevice.Handshake = Handshake.None;
+            _serialDevice.DataBits = 8;
+            _serialDevice.ReadBufferSize = 2048;
+            _serialDevice.ReadTimeout = 2000;
+            _serialDevice.WatchChar = '\r';
+            _serialDevice.DataReceived += SerialDevice_DataReceived;
```

Comment:
open question: do you want to add the event before starting? And remove them after stopping?

Reply:
Is there any difference? It should not be invoked without start. But maybe it should be moved into Open method?
So add after start and remove before stop?

Reply:
yes, I think it's better to add an remove at start and stop, things stays clean then

Reply:
done